src/apirequest.ts

```diff
@@ -167,13 +167,10 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
   }
 
   // Rewrite url if rootUrl is globally set
-  if (
```

Comment:
@bcoe @sofisl this is a very sensitive area - if the PR looks good, before merging please do a clone, link, and make sure to run the unit tests for google-api-nodejs-client (or merge, and do the clone/link against the un-released library)

Reply:
@JustinBeckwith any news?

Reply:
@yitomok I haven't been able to do the end to end testing required to merge this PR.

Appreciate the contribution, and we will work on landing it soon. As @JustinBeckwith says, I need to fully understand how this will affect our libraries, and do some testing, before we merge.

Reply:
@bcoe any news so far?

Reply:
@bcoe anything I could help to move forward?

Reply:
@yitomok, I've tested this PR by linking your fork to our `google-api-nodejs-client` library and attempting to run samples with `rootUrl` set.

I'm not sure if this PR solves the underlying issue. This code

```
const {GoogleApis} = require('googleapis');
const google = new GoogleApis({
  rootUrl: 'example.com',
});
const gmail = google.gmail('v1');
```  

Still returns, instead of throwing an error.

Reply:
@sofisl what do you mean "throwing an error"? This PR aims to fix on some levels (as the inline comment by their creators) are not respected/used at all. There is no validation on URL before or after.